src/helpers/augmentedReality/downloadObject.js

```diff
@@ -13,16 +13,27 @@ export const downloadObject = async ({ index, data, setData }) => {
   for (const objectItem of dataItem?.downloadableUris) {
     const { uri, title, type, id } = objectItem;
 
-    const storageName = storageNameCreator({ dataItem, objectItem });
+    const { directoryName, folderName, storageName } = storageNameCreator({ dataItem, objectItem });
 
     const downloadResumable = FileSystem.createDownloadResumable(
       uri,
-      FileSystem.cacheDirectory + storageName,
+      directoryName,
```

Comment:
what if we would use different `storageName`s instead of several directories? would it be possible to prefix `storageName`s with a timestamp or unique id to be always different? this way we could avoid the directory logics i think?

Reply:
it's hard to escape the logic of the index. Each texture file has a specific name. Viro needs this name to display a model. If we change the name of the texture file, the 3D object cannot be displayed properly.